server/controllers/Parcel.js

```diff
@@ -0,0 +1,168 @@
+    }
+  }
+
+  /**
+   * method to fetch particular parcel order
+   * @param {object} request 
+   * @param {object} response 
+   * @returns {object} particular parcel
+   */
+  static async getParcelById(request, response){
+    const getParcelQuery = 'SELECT * FROM parcels WHERE id=$1';
+    try{
+      const { rows, rowCount } = await db.query(getParcelQuery, [request.params.id]);
+      if( rowCount < 1 ){
+        return response.status(404).json({'Status':'404', 'message':' Order not found'});
+      }
+      return response.status(200).json({'status': 200, 'Data': rows[0]}) ;     
+    }
+    catch(error){
+      return response.status(400).json({'Status': 400, 'Error': `${error}`});
+    }
+  }
+
+  /**
+   * method to cancel a parcel delivery order
+   * @param {object} request 
+   * @param {object} response 
+   * @returns {object} parcel orders
+   */
+
+  static async cancelParcelOrder(request, response){
+    const findParcelQuery = 'SELECT * FROM parcels WHERE id = $1 AND placed_by=$2';
+    const updateParcelQuery = `UPDATE parcels SET cancelled=true, 
+          modified_at=NOW() WHERE id=$1 returning *`;
+    try{
+        const { rows, rowCount } = await db.query(findParcelQuery, [request.params.id, request.user.id]);
+      if(rowCount < 1){
+        return response.status(404).json({'Status': 404,'Message': 'No order with the specified id exists for the user'});
+      }
+
+      const result = await db.query(updateParcelQuery, [request.params.id]);
+      return response.status(200).json({'Status': 200,'Data': result.rows[0]});
+  
+    }
+    catch(error){
+      return response.status(400).json({'Status': 400, 'Error': `${error}`});
+    }
+  }
+
+  /**
+   * method to update parcel delivery location
+   * @param {object} request 
+   * @param {object} response 
+   * @returns {object} parcel orders
+   */
+
+   static async updateCurrentLocation(request, response){
+
+    const findParcelQuery = 'SELECT * FROM parcels WHERE id = $1';
+    const updateParcelQuery = `UPDATE parcels SET current_location=$1, 
+          modified_at=NOW() WHERE id=$2 returning *`;
+
+    const values = [
+      request.body.current_location,
+      request.params.id
+    ];
+    try{
+        const { rowCount } = await db.query(findParcelQuery, [request.params.id]);
+      if(rowCount < 1){
+        return response.status(404).json({'Status': 404,'Message': 'Order not found'});
+      }
+
+      const result = await db.query(updateParcelQuery, values);
+      return response.status(200).json({'Status': 200,'Message':'Location updated successfully','Data': result.rows[0]});
+  
+    }
+    catch(error){
+      return response.status(400).json({'Status': 400, 'Error': `${error}`});
+    }
+   }
+
+  /**
+   * method to update parcel delivery location
+   * @param {object} request 
+   * @param {object} response 
+   * @returns {object} parcel orders
+   */
+
+   static async updatedeDestination(){
+     
```

Comment:
why is this function empty?

Reply:
I was working on it last night and I slept off

Reply:
will implement it now